java/vector/src/test/java/org/apache/arrow/vector/ipc/TestArrowReaderWriter.java

```diff
@@ -135,4 +151,128 @@ public void test() throws IOException {
     }
   }
 
+  @Test
+  public void testWriteReadWithDictionaries() throws IOException {
+    DictionaryProvider.MapDictionaryProvider provider = new DictionaryProvider.MapDictionaryProvider();
+    provider.put(dictionary);
+
+    VarCharVector vector1 = newVarCharVector("varchar1", allocator);
+    vector1.allocateNewSafe();
+    vector1.set(0, "foo".getBytes(StandardCharsets.UTF_8));
+    vector1.set(1, "bar".getBytes(StandardCharsets.UTF_8));
+    vector1.set(3, "baz".getBytes(StandardCharsets.UTF_8));
+    vector1.set(4, "bar".getBytes(StandardCharsets.UTF_8));
+    vector1.set(5, "baz".getBytes(StandardCharsets.UTF_8));
+    vector1.setValueCount(6);
+    FieldVector encodedVector1 = (FieldVector) DictionaryEncoder.encode(vector1, dictionary);
+    vector1.close();
+
+    VarCharVector vector2 = newVarCharVector("varchar2", allocator);
+    vector2.allocateNewSafe();
+    vector2.set(0, "bar".getBytes(StandardCharsets.UTF_8));
+    vector2.set(1, "baz".getBytes(StandardCharsets.UTF_8));
+    vector2.set(2, "foo".getBytes(StandardCharsets.UTF_8));
+    vector2.set(3, "foo".getBytes(StandardCharsets.UTF_8));
+    vector2.set(4, "foo".getBytes(StandardCharsets.UTF_8));
+    vector2.set(5, "bar".getBytes(StandardCharsets.UTF_8));
+    vector2.setValueCount(6);
+    FieldVector encodedVector2 = (FieldVector) DictionaryEncoder.encode(vector2, dictionary);
+    vector2.close();
+
+    List<Field> fields = Arrays.asList(encodedVector1.getField(), encodedVector2.getField());
+    List<FieldVector> vectors = Collections2.asImmutableList(encodedVector1, encodedVector2);
+    VectorSchemaRoot root =  new VectorSchemaRoot(fields, vectors, encodedVector1.getValueCount());
+
+    ByteArrayOutputStream out = new ByteArrayOutputStream();
+    ArrowFileWriter writer = new ArrowFileWriter(root, provider, newChannel(out));
+    writer.start();
+    writer.writeBatch();
+    writer.end();
+
+    try (SeekableReadChannel channel = new SeekableReadChannel(
+        new ByteArrayReadableSeekableByteChannel(out.toByteArray()));
+        ArrowFileReader reader = new ArrowFileReader(channel, allocator)) {
+      Schema readSchema = reader.getVectorSchemaRoot().getSchema();
+      assertEquals(root.getSchema(), readSchema);
+      assertEquals(1, reader.getDictionaryBlocks().size());
+      assertEquals(1, reader.getRecordBlocks().size());
+
+      reader.loadNextBatch();
+      assertEquals(2, reader.getVectorSchemaRoot().getFieldVectors().size());
+    }
+  }
+
+  @Test
+  public void testEmptyStream1() throws IOException {
```

Comment:
please be more description in the test name for this and the one below.

Reply:
Thanks, fixed.